tests/integration_test.rs

```diff
@@ -466,3 +467,155 @@ async fn test_jsr_version_not_found_then_found() {
     ]
   );
 }
+
+#[tokio::test]
+async fn test_dynamic_imports_with_template_arg() {
+  async fn run_test(
+    code: &str,
+    files: Vec<(&str, &str)>,
+    expected_specifiers: Vec<&str>,
+  ) {
+    let mut loader = MemoryLoader::default();
+    let mut file_system = MemoryFileSystem::default();
+    for (specifier, text) in &files {
+      file_system.add_file(ModuleSpecifier::parse(specifier).unwrap());
+      loader.add_source_with_text(specifier, text);
+    }
+    loader.add_source_with_text("file:///main.ts", code);
+    let mut graph = ModuleGraph::new(GraphKind::All);
+    graph
+      .build(
+        vec![Url::parse("file:///main.ts").unwrap()],
+        &mut loader,
+        BuildOptions {
+          file_system: Some(&file_system),
+          ..Default::default()
+        },
+      )
+      .await;
+    graph.valid().unwrap();
+
+    let specifiers = graph
+      .specifiers()
+      .map(|s| s.0.to_string())
+      .filter(|s| s != "file:///main.ts")
+      .collect::<Vec<_>>();
+    assert_eq!(specifiers, expected_specifiers);
+  }
+
+  // relative with ./
+  run_test(
+    "
+await import(`./${test}`);
+",
+    vec![
+      ("file:///a/mod.ts", ""),
+      ("file:///a/sub_dir/a.ts", ""),
+      ("file:///b.ts", ""),
+    ],
+    vec!["file:///a/mod.ts", "file:///a/sub_dir/a.ts", "file:///b.ts"],
+  )
+  .await;
```

Comment:
To be honest I don't understand this bit - I assume you are returning all files that "might" match depending on the value of `test` variable - will this be handled somehow downstream or is the goal to just discover all possible files that might get imported?

Reply:
I think it works like in vite where it treats variables in the string as a `**` glob so it would include any possible file.

Reply:
I believe it uses a * glob instead of a ** glob.

For reference, the [Vite documentation](https://vitejs.dev/guide/features.html#dynamic-import) notes that variables only represent file names one level deep.

Additionally, Vite utilizes @rollup/plugin-dynamic-import-vars, which is described [here](https://github.com/rollup/plugins/tree/master/packages/dynamic-import-vars#globs-only-go-one-level-deep). According to the documentation:

> When generating globs, each variable in the string is converted to a glob * with a maximum of one star per directory depth. This avoids unintentionally adding files from many directories to your import.

Reply:
for example: 
```typescript
// /dev/main.ts
await import(`./b/${test}.ts`)
```
should not include: `file:///dev/b/c/c/test.ts`

Reply:
We ended up going with this behaviour here that's similar to vite, but not exactly.